kasa/tests/test_discovery.py

```diff
@@ -163,6 +165,40 @@ async def test_discover_single_hostname(discovery_mock, mocker):
         x = await Discover.discover_single(host, credentials=Credentials())
 
 
+async def test_discovery_credentials(mocker):
+    """Make sure that discover gives credentials precedence over un and pw."""
+    host = "127.0.0.1"
+    mocker.patch("kasa.discover._DiscoverProtocol.wait_for_discovery_to_complete")
+
+    def mock_discover(self, *_, **__):
+        self.discovered_devices = {host: MagicMock()}
+
+    mocker.patch.object(_DiscoverProtocol, "do_discover", mock_discover)
+    dp = mocker.spy(_DiscoverProtocol, "__init__")
+
+    await Discover.discover_single(host, credentials=Credentials(), timeout=0)
```

Comment:
This is testing multiple different things? I'd consider either parametrizing, or at least separate the checks to logical blocks with a comment.

Reply:
Split into separate tests for `discover` and `discover_single` and added some comments.